Subject: video: fix circular include in gstvideopool.h vs gstvideodmabufpool.h

Declare the bufferpool types before the video.h aggregate include that gstvideodmabufpool.h depends on.

--- a/gst-libs/gst/video/gstvideopool.h
+++ b/gst-libs/gst/video/gstvideopool.h
@@ -22,7 +22,7 @@
 
 #include <gst/gst.h>
 
-#include <gst/video/video.h>
+#include <gst/video/video-prelude.h>
 
 G_BEGIN_DECLS
 
@@ -45,14 +49,6 @@
  */
 #define GST_BUFFER_POOL_OPTION_VIDEO_ALIGNMENT "GstBufferPoolOptionVideoAlignment"
 
-/* setting a bufferpool config */
-
-GST_VIDEO_API
-void             gst_buffer_pool_config_set_video_alignment  (GstStructure *config, const GstVideoAlignment *align);
-
-GST_VIDEO_API
-gboolean         gst_buffer_pool_config_get_video_alignment  (const GstStructure *config, GstVideoAlignment *align);
-
 /* video bufferpool */
 typedef struct _GstVideoBufferPool GstVideoBufferPool;
 typedef struct _GstVideoBufferPoolClass GstVideoBufferPoolClass;
@@ -85,4 +81,18 @@
 
 G_END_DECLS
 
+#include <gst/video/video.h>
+
+G_BEGIN_DECLS
+
+/* setting a bufferpool config */
+
+GST_VIDEO_API
+void             gst_buffer_pool_config_set_video_alignment  (GstStructure *config, const GstVideoAlignment *align);
+
+GST_VIDEO_API
+gboolean         gst_buffer_pool_config_get_video_alignment  (const GstStructure *config, GstVideoAlignment *align);
+
+G_END_DECLS
+
 #endif /* __GST_VIDEO_POOL_H__ */
